src/HH4b/boosted/TrainBDT.py

```diff
@@ -44,28 +45,95 @@
+    ShapeVar(var="H2PNetMass", label=r"$m_{reg}^{2}$ (GeV)", bins=[30, 0, 300]),
+    ShapeVar(var="HHPt", label=r"HH $p_{T}$ (GeV)", bins=[30, 0, 4000]),
+    ShapeVar(var="HHEta", label=r"HH $\eta$", bins=[30, -5, 5]),
+    ShapeVar(var="HHMass", label=r"HH mass (GeV)", bins=[30, 0, 1500]),
+    ShapeVar(var="MET", label=r"MET (GeV)", bins=[30, 0, 600]),
+    ShapeVar(var="H1T32", label=r"$\tau_{32}^{0}$", bins=[30, 0, 1]),
+    ShapeVar(var="H2T32", label=r"$\tau_{32}^{1}$", bins=[30, 0, 1]),
+    ShapeVar(var="H1Pt", label=r"H $p_{T}^{0}$ (GeV)", bins=[30, 200, 1000]),
+    ShapeVar(var="H2Pt", label=r"H $p_{T}^{1}$ (GeV)", bins=[30, 200, 1000]),
+    ShapeVar(var="H1Eta", label=r"H $\eta^{0}$", bins=[30, -4, 4]),
+    ShapeVar(var="H1QCDb", label=r"QCDb$^{1}$", bins=[30, 0, 1]),
+    ShapeVar(var="H1QCDbb", label=r"QCDbb$^{1}$", bins=[30, 0, 1]),
+    ShapeVar(var="H1QCDothers", label=r"QCDothers$^{1}$", bins=[30, 0, 1]),
+    ShapeVar(var="H1Pt_HHmass", label=r"H$^0$ $p_{T}/mass$", bins=[30, 0, 1]),
+    ShapeVar(var="H2Pt_HHmass", label=r"H$^1$ $p_{T}/mass$", bins=[30, 0, 0.7]),
+    ShapeVar(var="H1Pt/H2Pt", label=r"H$^0$/H$^1$ $p_{T}$ (GeV)", bins=[30, 0.5, 1]),
+]
+
+# do not include small qcd bins
+for year in samples_run3:
+    samples_run3[year]["qcd"] = [
+        "QCD_HT-1000to1200",
+        "QCD_HT-1200to1500",
+        "QCD_HT-1500to2000",
+        "QCD_HT-2000",
+        # "QCD_HT-200to400",
+        "QCD_HT-400to600",
+        "QCD_HT-600to800",
+        "QCD_HT-800to1000",
+    ]
+
+
+def get_legtitle(legacy, pnet_xbb_str):
     title = r"FatJet p$_T^{(0,1)}$ > 300 GeV" + "\n" + "$T_{Xbb}^{0}$>0.8"
+    if "Legacy" in pnet_xbb_str:
+        title += "\n" + "PNet Legacy"
 
-    if not legacy:
-        title += "\n" + r"m$_{SD}^{(0,1)}$:[30-250] GeV"
+    if legacy:
+        title += "\n" + r"m$_{SD}$ or m$_{reg Legacy}$ > 50 GeV"
     else:
-        title += "\n" + r"m$_{Reg}^{(0,1)}$:[60-250] GeV"
+        title += "\n" + r"m$_{SD}$ or m$_{reg}$ > 50 GeV"
+        # title += "\n" + r"m$_{SD}$ > 30 GeV"
 
     return title
 
 
+def apply_cuts(events_dict, pnet_xbb_str, pnet_mass_str):
+    """
+    Apply cuts
+    Skimmer selection already includes
+    - 2 AK8 jets pT > 250 GeV, mSD > 60 or mReg > 60
+    - HLT OR
+    - Here we apply a pT selection at the very least
+    """
+    mass_cut = False
+    for key in events_dict:
+        msd1 = events_dict[key]["bbFatJetMsd"][0]
+        msd2 = events_dict[key]["bbFatJetMsd"][1]
+        pt1 = events_dict[key]["bbFatJetPt"][0]
+        pt2 = events_dict[key]["bbFatJetPt"][1]
+        xbb1 = events_dict[key][pnet_xbb_str][0]
+        mass1 = events_dict[key][pnet_mass_str][0]
+        mass2 = events_dict[key][pnet_mass_str][1]
+        events_dict[key] = events_dict[key][(pt1 > 300) & (pt2 > 300) & (xbb1 > 0.8)].copy()
+        if mass_cut:
+            events_dict[key] = events_dict[key][(msd1 > 30) & (msd2 > 30)].copy()
```

Comment:
Do we want this 30 gev cut on both jets for legacy as well?

Reply:
I am not sure we do, would that get rid of the low mSD jets in signal too?

Reply:
Yeah it would. Maybe I misunderstood the code though - I thought this was applying the same 30 gev cut for both legacy and v12, whereas I thought we only needed the cut for v12.

Reply:
And I guess for jet 1 / the jet which gets triggered on.